src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -365,15 +365,15 @@ pub(crate) fn get_tool_rustc_compiler(
     builder.compiler(target_compiler.stage.saturating_sub(1), builder.config.host_target)
 }
 
-/// Returns a compiler that is able to compile a `ToolTarget` tool for the given `target`.
-pub(crate) fn get_tool_target_compiler(builder: &Builder<'_>, target: TargetSelection) -> Compiler {
-    todo!("FIX");
-    if builder.host_target == target {
+/// Returns the smallest stage compiler that is able to compile code for the given `target`.
+pub(crate) fn get_compiler_for_target(builder: &Builder<'_>, target: TargetSelection) -> Compiler {
+    let compiler = if builder.host_target == target {
         builder.compiler(0, builder.host_target)
     } else {
-        // FIXME: should this be builder.top_stage to avoid rebuilds?
-        builder.compiler(1, target)
-    }
+        builder.compiler(1, builder.host_target)
+    };
+    builder.std(compiler, target);
+    compiler
```

Comment:
Question [STAGING 2/2]: *oh*, it's changed in this commit. But wait, doesn't this still disagree with the comment re. building `ToolTarget` tools with the in-tree compiler and std, even when not cross-compiling?